src/export_data.py

```diff
@@ -0,0 +1,122 @@
+import ee
+from utils import TaskManager, SampleExporter, stack_bands_from_imagery
+from utils import RoadImagery, MultiSpectralImagery, TropomiImagery
+from utils import DSMImagery, WindImagery
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Export Data from EarthEngine'
+                                     )
+    parser.add_argument('--params_path', type=str,
+                        help="json parameters file location")
+    parser.add_argument('--export_folder', type=str,
+                        help="Cloud export folder")
+    parser.add_argument('--bucket', type=str,
+                        help="bucket name",
+                        default='aghriss-air-quality')
+    parser.add_argument('--num_samples', type=int,
+                        help="number of samples to export per image",
+                        default=1000)
+    parser.add_argument('--num_shards', type=int,
+                        help="number of sharding of samples before export",
+                        default=10)
+
+    args = parser.parse_args()
+
+    params = json.load(open(args.params_path, 'r'))
+    BUCKET = args.bucket
+    DIRECTORY = args.export_folder
+    num_samples = args.num_samples
+    num_shards = args.num_shards
+
+    print("Loading", (args.params_path))
+    params_path = args.params_path
+    params = json.load(open(params_path, 'r'))
+    PATCH_BANDS = []
+    PATCH_BANDS += params['bands']['multispectral']
+    PATCH_BANDS += params['bands']['tropomi']
+    PATCH_BANDS += params['bands']['road'] + params['bands']['dsm']
+    STACKED_WIND_BANDS = ["%i_%s" % (i, b) for b in params['bands']['wind']
+                          for i in range(12)]
+    PATCH_BANDS += STACKED_WIND_BANDS
+    BANDS = PATCH_BANDS + ["HOD", "DOW", "DOM", "MOY", "latitude", "longitude",
+                           "valid"]
+    SCALE = params['scale']
+    KERNEL_RADIUS = params['kernel_radius']
+    # Initialize EE
+    ee.Authenticate()
+    ee.Initialize()
+    # Creating diffrent kernels
+    kernel = ee.Kernel.square(KERNEL_RADIUS)
+    # vertical and horizontal kernel used in calculating valid pixels
+    vertical_kernel = ee.Kernel.rectangle(xRadius=1,
+                                          yRadius=KERNEL_RADIUS,
+                                          units='pixels')
+    horizontal_kernel = ee.Kernel.rectangle(xRadius=KERNEL_RADIUS,
+                                            yRadius=1,
+                                            units='pixels')
+    # Initializing smaple exporter and TaskManager
+    task_manager = TaskManager(verbose=True)
+    sampler = SampleExporter(task_manager, num_samples, num_shards,
+                             kernel, SCALE, BUCKET, DIRECTORY)
+    # Initializing imagery classes
+    multispectral = MultiSpectralImagery(params['collections']['multispectral'],
+                                         start_date="2018-01-01",
+                                         end_date="2019-12-31",
```

Comment:
Do we want to add flags for these dates? Or do we expect these dates to never change?

Reply:
That might be useful for future use. For this project the date range is fixed